core/src/main/java/org/infinispan/interceptors/compat/TypeConverterStream.java

```diff
@@ -0,0 +1,66 @@
+package org.infinispan.interceptors.compat;
+
+import org.infinispan.CacheStream;
+import org.infinispan.commons.util.IteratorMapper;
+import org.infinispan.compat.TypeConverter;
+import org.infinispan.container.InternalEntryFactory;
+import org.infinispan.container.entries.CacheEntry;
+import org.infinispan.stream.impl.AbstractDelegatingCacheStream;
+import org.infinispan.stream.impl.spliterators.IteratorAsSpliterator;
+
+import java.util.Iterator;
+import java.util.Spliterator;
+
+/**
+ * Delegating stream that converts variables.  Note this stream specifically doesn't specify generics as it can have
```

Comment:
It isn't specific to keys and values though, since this can be used for other types besides CacheEntry like just values or whatever a map method transformed it into  :)